### DISCUSSION TITLE
Layer authentication in Hajk

### DISCUSSION THREAD

**Discussion by Abelon38** [hajkmap/Hajk#1477] [Q&A]
Hi guys!

I wonder if there is a way to add layers with credentials in Hajk? We are using QGIS-server for Open-data and Geoserver for data that requires credentials. Directly in the Hajk-admin I dont see thart possibility. Can someone light me up with this question?

Thank you for your time!

Abel

**Comment by jacobwod** [comment]
Supplying the credentials in Hajk's Admin would expose them in layers.json, which is fetched by client and easily accessible for anyone.

The way I've solved it in our organisation is using a reverse proxy and supply the credentials there. You can use any solution you wish, or just go for the easy option which is built-in into the NodeJS backend. See:https://github.com/hajkmap/Hajk/blob/b0ef09fd94b0c100b10fd9d0ec5eb1d2fe0365d6/new-backend/.env#L118

Depending on your setup (i.e. if the server accepts the credentials as basic HTTP query params), you could do something like:
```
PROXY_QGISSERVER=https://qgis.example.com/wms-service/?user=abc&pass=1234
```

**Reply by Abelon38** [reply]
Thanks Jacob! I will give it a try using PROXY_QGISSERVER=https://qgis.example.com/wms-service/?user=abc&pass=1234 I will come back with the result. 

have a great evening! 🙂
